src/syntax/parser.rs

```diff
@@ -320,17 +320,33 @@ fn math_expr_prec(p: &mut Parser, min_prec: usize, stop: SyntaxKind) {
             let m3 = p.marker();
             math_expr_prec(p, prec, SyntaxKind::Eof);
             math_unparen(p, m3);
+        } else if p.current_text() == "("
```

Comment:
Unless there is a problem with it, I think this should be handled through the normal precedence mechanism instead of this ad-hoc solution.

Reply:
I'll be honestly I'm struggling to understand the parser code. If you can point me in the right direction (or if there are docs or conversations in discord I can read), that would be super helpful.

Basic questions: does `prec` stand for precedence? What is `min_prec` for and why is `3` a magic number used in some of the if statements?

Reply:
Awesome, thanks for the explanation! Currently a little swamped, so I won't have time to fix this for another few weeks.